app/assets/javascripts/views/shared/chartWidgetView.js

```diff
@@ -253,7 +255,36 @@
       // We append the button to the DOM
       this.el.querySelector('.js-switch-button').appendChild(button);
     },
+    /**
+     * Render the Toggle Visibility button and attach an event listener to it
+     */
+    _renderToggleVisibilityButton: function () {
+      // We create the button
+      var button = document.createElement('button');
+      button.type = 'button';
+      button.classList.add('c-button', 'toggle-visibility-button');
+      button.textContent = 'Hide';
+      // We attach the listener
+      button.addEventListener('click', () => {
+        this._toggleWidgetVisibility();
+        this.options.toggleVisibilityCallback();
+        this._renderChart();
+      });
 
+      // We append the button to the DOM
+      this.el.querySelector('.js-toggle-visibility-button').appendChild(button);
+    },
+    _toggleWidgetVisibility: function(){
+      const visibleButton = this.el.querySelector('.toggle-visibility-button');
+      const largeWidget = visibleButton.closest('.large-widget');
```

Comment:
`closest` method is not supported in all major browsers, use a different selector or fallback to jQuery. We're already using it in other views in this project.